app/src/main/java/explorer/city/com/cityexplorer/view/adapter/viewHolder/SearchViewHolder.kt

```diff
@@ -2,14 +2,25 @@ package explorer.city.com.cityexplorer.view.adapter.viewHolder
 
 import android.support.v7.widget.RecyclerView
 import android.view.View
+import com.bumptech.glide.Glide
+import com.bumptech.glide.TransitionOptions
 import explorer.city.com.cityexplorer.model.SearchItem
 import explorer.city.com.cityexplorer.view.listener.OnItemClickListener
 import kotlinx.android.synthetic.main.list_item__search_result.view.*
 
 class SearchViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
 
-    fun setSearchResult(searchItem: SearchItem): SearchViewHolder {
-        itemView.name.text = searchItem.fullName
+    fun setCityName(cityName: String): SearchViewHolder {
+        itemView.name.text = cityName
+        return this
+    }
+
+    fun setCityImage(imageUrl: String?): SearchViewHolder {
+        imageUrl?.let {
+            Glide.with(itemView.cityImage.context)
```

Comment:
Here you have the opportunity to use a class extension, so that you can move Glide-specific code away from the view layer.